cabal.project.stage2

```diff
@@ -71,6 +76,40 @@ tests: False
 allow-boot-library-installs: True
 active-repositories: :none
 
+if arch(javascript)
+  constraints:
+    Cabal source,
+    Cabal-syntax source,
+    array source,
+    base source,
+    binary source,
+    bytestring source,
+    containers source,
+    deepseq source,
+    directory source,
+    exceptions source,
+	file-io source,
+    filepath source,
+    hpc source,
+    integer-gmp source,
+    mtl source,
+	os-string source,
+    parsec source,
+    pretty source,
+    process source,
+    rts source,
+    rts-headers source,
+	rts-fs source,
+    stm source,
+    system-cxx-std-lib source,
+    template-haskell source,
+    text source,
+    time source,
+    transformers source,
+    unix source,
+    xhtml source,
+	Win32 source
```

Comment:
The indentation is meant to signify dependencies?

Why do we need the source constraint here? We should have acomment explaining why this is done, and if this is a cabal defect that needs to be rectified or something else.